RadixWalletTests/Features/AssetTransferTests/AssetTransferDepositRuleTests.swift

```diff
@@ -24,120 +24,167 @@ final class AssetTransferDepositRuleTests: TestCase {
+	func test__GIVEN__depositRuleIsAcceptKnown__WHEN__accountContainResource_resourceIsInAllowList__THEN__noSignatureIsRequired() async throws {
 		var account = Self.recipientAccount
+		// GIVEN
 		account.onLedgerSettings.thirdPartyDeposits.depositRule = .acceptKnown
+		// WHEN
 		account.onLedgerSettings.thirdPartyDeposits.assetsExceptionList = [.init(address: Self.resourceAddress, exceptionRule: .allow)]
 
-		await assertNoSignatureIsRequired(for: account, onLedgerAccounts: [Self.onLedgerAccountWithResource])
+		// THEN
+		try await assertNoSignatureIsRequired(for: account, onLedgerAccounts: [Self.onLedgerAccountWithResource])
 	}
 
-	func test_GIVEN_depositRuleIsAcceptKnown_WHEN_accountContainResource_resourceIsInDenyList_THEN_signatureIsRequired() async {
+	func test__GIVEN__depositRuleIsAcceptKnown__WHEN__accountContainResource_resourceIsInDenyList__THEN__signatureIsRequired() async throws {
 		var account = Self.recipientAccount
+		// GIVEN
 		account.onLedgerSettings.thirdPartyDeposits.depositRule = .acceptKnown
+		// WHEN
 		account.onLedgerSettings.thirdPartyDeposits.assetsExceptionList = [.init(address: Self.resourceAddress, exceptionRule: .deny)]
 
-		await assertSignatureIsRequired(for: account, onLedgerAccounts: [Self.onLedgerAccountWithResource])
+		// THEN
+		try await assertSignatureIsRequired(for: account, onLedgerAccounts: [Self.onLedgerAccountWithResource])
 	}
 
 	// MARK: - Private
 
-	private func assertSignatureIsRequired(for account: Profile.Network.Account, onLedgerAccounts: [OnLedgerEntity.Account] = []) async {
-		await assertSignatureIsRequired(for: account, isRequired: true, onLedgerAccounts: onLedgerAccounts)
+	private func assertSignatureIsRequired(
+		for account: Profile.Network.Account,
+		onLedgerAccounts: [OnLedgerEntity.Account] = []
+	) async throws {
+		try await assertSignatureIsRequired(for: account, isRequired: true, onLedgerAccounts: onLedgerAccounts)
 	}
 
-	private func assertNoSignatureIsRequired(for account: Profile.Network.Account, onLedgerAccounts: [OnLedgerEntity.Account] = []) async {
-		await assertSignatureIsRequired(for: account, isRequired: false, onLedgerAccounts: onLedgerAccounts)
+	private func assertNoSignatureIsRequired(
+		for account: Profile.Network.Account,
+		onLedgerAccounts: [OnLedgerEntity.Account] = []
+	) async throws {
+		try await assertSignatureIsRequired(for: account, isRequired: false, onLedgerAccounts: onLedgerAccounts)
 	}
 
-	private func assertSignatureIsRequired(for account: Profile.Network.Account, isRequired: Bool, onLedgerAccounts: [OnLedgerEntity.Account]) async {
-		await withDependencies { d in
-			d.onLedgerEntitiesClient.getEntities = { _, _, _, _ in
-				onLedgerAccounts.map { .account($0) }
+	private func assertSignatureIsRequired(for account: Profile.Network.Account, isRequired: Bool, onLedgerAccounts: [OnLedgerEntity.Account]) async throws {
+		try await withTimeLimit(.fast) {
```

Comment:
@GhenadieVP hmm I think best to have it in each test... like I did, else one might do async stuff later in test, and that would defeat the purpose of this.